adal/src/Microsoft.IdentityModel.Clients.ActiveDirectory/Platforms/net45/PlatformParameters.cs

```diff
@@ -49,6 +52,7 @@ public PlatformParameters(PromptBehavior promptBehavior):this(promptBehavior, nu
         /// <param name="ownerWindow"></param>
         public PlatformParameters(PromptBehavior promptBehavior, object ownerWindow)
         {
+            _syncContext = SynchronizationContext.Current ?? new SynchronizationContext();
```

Comment:
>_syncContext [](start = 12, length = 12)

Does this enforce the ui thread context? For other libraries we have put the synhronization responsiblities on the caller, however naturally we have to open the  web ui on the UI thread, that should be done where the web view is created though (assuming the controls are not created on the ui thread)

Reply:
So this is assuming that the PlatformParameters will be created on the UI thread. Which makes sense if you are passing a window object anyway. However, ownerWindow is only used to find another window and to center the browser UI onto it. Most people will pass in null (and so do our samples). So, it is quite possible that we users will create PlatformParamters on a non-UI thread.

I just wonder if we should be more explicit to our users, perhaps through the PlatformParamters comments (which are missing completely) or better yet via another constructor that takes in a syncronization context ? 

CCing @jmprieur for documentation implications..